Documentation/diagnostics/source_link.md

```diff
@@ -0,0 +1,137 @@
+  </Target>
+...
+```
+
+### Source Link JSON Schema
+```json
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "SourceLink",
+    "description": "A mapping of source file paths to URL's",
+    "type": "object",
+    "properties": {
+        "documents": {
+            "type": "object",
+            "minProperties": 1,
+            "additionalProperties": {
+                "type": "string"
+            },
+            "description": "Each document is defined by a file path and a URL. Original source file paths are compared 
+                             case insensitive to documents and the resulting URL is used to download source. The document 
+                             may contain an asterisk to represent a wildcard in order to match anything in the asterisk's 
+                             location. The rules for the asterisk are as follows:
+                             1. The only acceptable wildcard is one and only one '*', which if present will be replaced by a relative path.
+                             2. If the file path does not contain a *, the URL cannot contain a * and if the file path contains a * the URL must contain a *.
+                             3. If the file path contains a *, it must be the final character.
+                             4. If the URL contains a *, it may be anywhere in the URL."
+        }
+    },
+    "required": ["documents"]
+}
+```
+
+### Examples
+The simplest Source Link JSON would map every file in the project directory to a URL with a matching relative path. The following example shows what the Source Link JSON would look like for the .NET [Code Formatter](https://github.com/dotnet/codeformatter). In this example, Code Formatter has been cloned to `C:\src\CodeFormatter\` and is being built at commit `bcc51178e1a82fb2edaf47285f6e577989a7333f`.
+```json
+{
+    "documents": {
+        "C:\\src\\CodeFormatter\\*": "https://raw.githubusercontent.com/dotnet/codeformatter/bcc51178e1a82fb2edaf47285f6e577989a7333f/*"
+    }
+}
+```
+In this example, the file `C:\src\CodeFormatter\src\Microsoft.DotNet.CodeFormatting\FormattingEngine.cs` would be retrieved from https://raw.githubusercontent.com/dotnet/codeformatter/bcc51178e1a82fb2edaf47285f6e577989a7333f/src/Microsoft.DotNet.CodeFormatting/FormattingEngine.cs
+
+If the project was built on linux and was cloned to `/usr/local/src/CodeFormatter/`, the Source Link JSON would be:
+```json
+{
+    "documents": {
+        "/usr/local/src/CodeFormatter/*": "https://raw.githubusercontent.com/dotnet/codeformatter/bcc51178e1a82fb2edaf47285f6e577989a7333f/*"
+    }
+}
+```
+
+Source Link configuration can also contain absolute mappings from file paths to URLs without an asterisk. Using the Code Formatter example from above, every source file could be mapped explicitly:
+```
```

Comment:
Ah, well, in that case, you can use `js` instead and put it in a comment: `// ...`.